lib/dynamoid/criteria/key_fields_detector.rb

```diff
@@ -24,10 +24,11 @@ def contain_with_eq_operator?(field_name)
         end
       end
 
-      def initialize(query, source)
+      def initialize(query, source, forced_index_name)
```

Comment:
**suggestion:** I would prefer using keyword argument for optional parameters
```suggestion
      def initialize(query, source, forced_index_name: nil)
```

Reply:
I've updated all instances of the class usage to pass a value, albeit null. I wasn't sure if there would be a need to take in on initialization at some point in the future. This way all constructor calls are the same three instance vars.

I can revert that change to make it an optional keyword param if you prefer that to passing a known null value.